framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -3771,5 +3801,789 @@ std::wstring Dx12ReplayConsumerBase::ConstructObjectName(format::HandleId captur
+    Decoded_D3D12_CPU_DESCRIPTOR_HANDLE                          DestDescriptor)
+{
+    auto heap_object_info = GetObjectInfo(DestDescriptor.heap_id);
+    auto heap_extra_info  = GetExtraInfo<D3D12DescriptorHeapInfo>(heap_object_info);
+
+    RenderTargetInfo info;
+    info.resource_id   = pResource;
+    info.replay_handle = *DestDescriptor.decoded_value;
+    if (pDesc->IsNull())
+    {
+        info.is_view_null = true;
+    }
+    else
+    {
+        info.view         = *(pDesc->GetMetaStructPointer()->decoded_value);
+        info.is_view_null = false;
+    }
+
+    heap_extra_info->render_target_infos.emplace_back(std::move(info));
+}
+
+void Dx12ReplayConsumerBase::PostCall_ID3D12Device_CreateDepthStencilView(
+    const ApiCallInfo&                                           call_info,
+    DxObjectInfo*                                                object_info,
+    format::HandleId                                             pResource,
+    StructPointerDecoder<Decoded_D3D12_DEPTH_STENCIL_VIEW_DESC>* pDesc,
+    Decoded_D3D12_CPU_DESCRIPTOR_HANDLE                          DestDescriptor)
+{
+    auto heap_object_info = GetObjectInfo(DestDescriptor.heap_id);
+    auto heap_extra_info  = GetExtraInfo<D3D12DescriptorHeapInfo>(heap_object_info);
+
+    DepthStencilInfo info;
+    info.resource_id   = pResource;
+    info.replay_handle = *DestDescriptor.decoded_value;
+    if (pDesc->IsNull())
+    {
+        info.is_view_null = true;
+    }
+    else
+    {
+        info.view         = *(pDesc->GetMetaStructPointer()->decoded_value);
+        info.is_view_null = false;
+    }
+    heap_extra_info->depth_stencil_infos.emplace_back(std::move(info));
+}
+
+void Dx12ReplayConsumerBase::PostCall_ID3D12GraphicsCommandList_OMSetRenderTargets(
+    const ApiCallInfo&                                         call_info,
+    DxObjectInfo*                                              object_info,
+    UINT                                                       NumRenderTargetDescriptors,
+    StructPointerDecoder<Decoded_D3D12_CPU_DESCRIPTOR_HANDLE>* pRenderTargetDescriptors,
+    BOOL                                                       RTsSingleHandleToDescriptorRange,
+    StructPointerDecoder<Decoded_D3D12_CPU_DESCRIPTOR_HANDLE>* pDepthStencilDescriptor)
+{
+    if (call_info.index == track_dump_resources_.target.set_render_targets_code_index)
+    {
+        auto rt_handles = pRenderTargetDescriptors->GetMetaStructPointer();
+        track_dump_resources_.render_target_heap_ids.resize(NumRenderTargetDescriptors);
+        track_dump_resources_.replay_render_target_handles.resize(NumRenderTargetDescriptors);
+        for (uint32_t i = 0; i < NumRenderTargetDescriptors; ++i)
+        {
+            track_dump_resources_.render_target_heap_ids[i]       = rt_handles[i].heap_id;
+            track_dump_resources_.replay_render_target_handles[i] = *rt_handles[i].decoded_value;
+        }
```

Comment:
If `RTsSingleHandleToDescriptorRange == true` and `NumRenderTargetDescriptors > 1` then `pRenderTargetDescriptors` will only point the the first descriptor and will only have one valid meta struct. The other descriptors are contiguous in the descriptor heap. Here is a rough draft of a possible solution:

```
auto descriptor_increment = 0;
if (RTsSingleHandleToDescriptorRange)
{
    // Determine descriptor increment for contiguous render target descriptors.
    auto cmd_list = reinterpret_cast<ID3D12GraphicsCommandList*>(object_info->object);
    auto device   = graphics::dx12::GetDeviceComPtrFromChild<ID3D12Device>(cmd_list);
    if (device != nullptr)
    {
        descriptor_increment = device->GetDescriptorHandleIncrementSize(D3D12_DESCRIPTOR_HEAP_TYPE_RTV);
    }
    else
    {
        // ERROR
    }
}

for (uint32_t i = 0; i < NumRenderTargetDescriptors; ++i)
{
    if (RTsSingleHandleToDescriptorRange)
    {
        // All render target descriptors are contiguous in the same descriptor heap.
        track_dump_resources_.render_target_heap_ids[i] = rt_handles[0].heap_id;
        track_dump_resources_.replay_render_target_handles[i] =
            D3D12_CPU_DESCRIPTOR_HANDLE{ (*rt_handles[0].decoded_value).ptr + descriptor_increment };
    }
    else
    {
        track_dump_resources_.render_target_heap_ids[i]       = rt_handles[i].heap_id;
        track_dump_resources_.replay_render_target_handles[i] = *rt_handles[i].decoded_value;
    }
}
```

Some references:
- Notes on `RTsSingleHandleToDescriptorRange` here: https://learn.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12graphicscommandlist-omsetrendertargets
- Encode function which only encodes one descriptor if `RTsSingleHandleToDescriptorRange == true`: https://github.com/LunarG/gfxreconstruct/blob/278ceb7a7be02f7eeb928ff43af80f0674fc0088/framework/generated/generated_dx12_api_call_encoders.cpp#L5829

Reply:
Updated